include/exadg/incompressible_navier_stokes/spatial_discretization/spatial_operator_base.cpp

```diff
@@ -195,23 +195,18 @@ SpatialOperatorBase<dim, Number>::setup(
   // initialize data container for DirichletCached boundary conditions
   if(not(boundary_descriptor->velocity->dirichlet_cached_bc.empty()))
   {
-    interface_data_dirichlet_cached = std::make_shared<ContainerInterfaceData<1, dim, double>>();
     std::vector<unsigned int> quad_indices;
     quad_indices.emplace_back(get_quad_index_velocity_linear());
     quad_indices.emplace_back(get_quad_index_velocity_nonlinear());
     quad_indices.emplace_back(get_quad_index_velocity_gauss_lobatto());
 
-    std::set<dealii::types::boundary_id> bids;
-    for(auto const & bc : boundary_descriptor->velocity->dirichlet_cached_bc)
-      bids.insert(bc.first);
-
-    interface_data_dirichlet_cached->setup(matrix_free,
+    interface_data_dirichlet_cached = std::make_shared<ContainerInterfaceData<1, dim, double>>();
+    interface_data_dirichlet_cached->setup(*matrix_free,
                                            get_dof_index_velocity(),
                                            quad_indices,
-                                           bids);
+                                           boundary_descriptor->velocity->dirichlet_cached_bc);
```

Comment:
I guess you allocate the data structures for the complete boundary?

Reply:
Yes, the "boundary condition" is no longer a map with bid+function, but only a set of boundary ids (= dirichlet_cached_bc), since the underlying "Function" to evaluate is a purely internal data structure.